HW0/P4/P4.py

```diff
@@ -40,38 +45,106 @@
     # Read the observation array and plot it (Part 2)
     #####################
 
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '.g', label='Observed trajectory')
+    s_meas = np.loadtxt('P4_measurements.txt',delimiter=',')
+
+    s_meas_positions = s_meas[:,:3] # just the position information, Nx3 array
+
+    s_meas_positionsT = s_meas_positions.transpose() # convert to 3xN array
+
+    R = np.array([[1/rx,0,0],[0,1/ry,0],[0,0,1/rz]]) # instrument matrix 3x3
+
+    s_measCT = np.dot(R,s_meas_positionsT) # calibrate measurement
+    s_measC = s_measCT.transpose() # transpose back, because otherwise plotting somehow doesn't work..
+    
+    x_coords = s_measC[:,0]
+    y_coords = s_measC[:,1]
+    z_coords = s_measC[:,2]
+
+    ax.plot(x_coords, y_coords, z_coords,
+             '.g', label='Observed trajectory')
 
     #####################
     # Part 3:
     # Use the initial conditions and propagation matrix for prediction
     #####################
 
-    # A = ?
-    # a = ?
-    # s = ?
+    A = np.array([[1,0,0,dt,0,0],[0,1,0,0,dt,0],[0,0,1,0,0,dt],[0,0,0,1-c*dt,0,0],[0,0,0,0,1-c*dt,0],[0,0,0,0,0,1-c*dt]])
```

Comment:
I personally dislike populating "matrices" this way, especially relatively sparse matrices. I usually try to do things like this np.diag([1, 1, 1, 1-c*dt, 1-c*dt, 1-c*dt]) + np.diag([dt, dt ,dt],-3 ) because I think the method is kind of cool and it isn't as tedious.